src/components/DataControllerMixin/DataControllerMixin.js

```diff
@@ -51,6 +51,16 @@ var ModalMixin = {
         area: props.area || null,
         time: timeidx,
       },
+    }).then(function(data, textstatus, jqXHR) {
+      if($.isEmptyObject(data)) {
+        return $.Deferred().reject(jqXHR, "empty", textstatus);
+      }
+      for(var run in data) {
+        if(!('data' in data[run]) || !('units' in data[run])){
+          return $.Deferred().reject(jqXHR, "incomplete", textstatus);            
+        }
+      }
+      return $.Deferred().resolve(data, textstatus, jqXHR);
```

Comment:
@corviday, James has asked me to act on my [issue/recommendation about promises](https://github.com/pacificclimate/climate-explorer-frontend/issues/45). This will cause some changes (mainly simplifying) to your code here. Let's discuss.